frontend/src/components/smart-contracts/ERC20ApprovalProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import { Listener } from '@ethersproject/providers';
+import { BigNumber } from 'ethers';
+import {
+  SmartContractAddresses,
+  useSmartContracts,
+  SmartContractActions,
+} from 'components/smart-contracts/useSmartContracts';
+import { useWallet, useWalletFilter } from 'components/wallet/useWallet';
+
+export const ERC20ApprovalProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const wallet = useWallet();
+  const { contractsState, dispatch } = useSmartContracts();
+
+  // attach the filter handler
+  const onApprovalHandler: Listener = (owner, spender, approvalAmount) => {
```

Comment:
If this is a provider, shouldn't we just be defining functions and state that we're providing via context? Usually we would not do this stuff in a provider.

Reply:
Moved this into ContractsProvider.  Could be refactored further to pull the filter handling into it's own hook.

Reply:
I did refactor this to move the building functions into SmartContractContext